lib/git.zsh

```diff
@@ -94,11 +94,13 @@ function git_remote_status() {
 # Using '--quiet' with 'symbolic-ref' will not cause a fatal error (128) if
 # it's not a symbolic ref, but in a Git repo.
 function git_current_branch() {
-  local ref
-  ref=$(__git_prompt_git symbolic-ref --quiet HEAD 2> /dev/null)
+  local ref=$(__git_prompt_git branch --show-current 2> /dev/null)
   local ret=$?
```

Comment:
This $? won't work. It'll get the return code of the `local` command instead of the git one. That's why it's a two-line statement.

Reply:
I've updated the branch accordingly.

Reply:
That's fine, but my point about it being counterproductive/unnecessary still stands.

Reply:
If you (and the other maintainers) decide that it's too much overhead, it's your prerogative to decide not to use this PR. I'll accept your decision either way.